sdk/src/Core/Amazon.Runtime/Credentials/ProcessAWSCredentials.cs

```diff
@@ -176,6 +176,15 @@ private CredentialsRefreshState SetCredentialsRefreshState(ProcessExecutionResul
                         try
                         {
                             processCredentialDataV1 = JsonSerializerHelper.Deserialize<ProcessCredentialVersion1>(processInfo.StandardOutput, ProcessCredentialVersion1JsonSerializerContexts.Default);
+                            if (processCredentialDataV1.Expiration == DateTime.MaxValue && processCredentialDataV1.Expiration.Kind != DateTimeKind.Utc)
```

Comment:
why do we need to check if the expiration is DateTime.MaxValue? wouldn't we want to convert the expiration to universal time regardless of if it is MaxValue?

Reply:
The default for the ProcessCredentialVersion1 class is to set the expiration to DateTime.MaxValue. The problem with this is it will be an unspecified time. We want all times to be in UTC so we check if it is that default and if it isn't UTC we set the Kind as Utc. This retains the value of MaxValue and just updates the kind. We cannot do ToUniversal for example as it will go out of bounds of the DateTime type.